diff --git a/pkg/router/f5/f5.go b/pkg/router/f5/f5.go
index 71aa4233f7..c672b8310e 100644
--- a/pkg/router/f5/f5.go
+++ b/pkg/router/f5/f5.go
@@ -12,6 +12,7 @@ import (
 	"path"
 	"strings"
 	"regexp"
+	"time"
 
 	"github.com/golang/glog"
 	knet "k8s.io/apimachinery/pkg/util/net"
@@ -258,8 +259,16 @@ func (f5 *f5LTM) restRequest(verb string, url string, payload io.Reader, filesiz
 		// Example for a 1K file in one chunk: Content-Range: 0-1023/1024
 	}
 
+	// Disable redirect-following, so credentials are only sent to the known endpoint.
+	redirHandler := func(req *http.Request, via []*http.Request) error {
+		return http.ErrUseLastResponse
+	}
+
+	// Add a timeout to prevent deadlocks in case ie. a firewall starts dropping segments.
+	timeout := 60 * time.Second
+
 	// Create a HTTP client on top of the TLS transport.
-	client := &http.Client{Transport: tr}
+	client := &http.Client{Transport: tr, CheckRedirect: redirHandler, Timeout: timeout}
 
 	// Redact credentials from log output.
 	details := fmt.Sprintf("%v", req)
